drivers/auxiliary/cheapodc.cpp

```diff
@@ -112,61 +112,90 @@ bool CheapoDC::initProperties()
+    IUGetConfigText(getDeviceName(), "TEMPERATURE_SNOOP", "TEMPERATURE_Property", temperatureProperty, MAXINDINAME);
     IUGetConfigText(getDeviceName(), "TEMPERATURE_SNOOP", "TEMPERATURE_ATTRIBUTE", temperatureAttribute, MAXINDINAME);
 
-    if (IUGetConfigOnSwitchIndex(getDeviceName(), "ENABLE_SNOOP_LOCATION", &snoopLocationIndex) == 0)
-        setSnoopLocation = (snoopLocationIndex == 0);
-    if (IUGetConfigOnSwitchIndex(getDeviceName(), "ENABLE_SNOOP_TEMPERATURE", &snoopTemperatureIndex) == 0) 
-        setSnoopTemperature = (snoopTemperatureIndex == 0);
+   // TEMPERATURE_Property text case kept for backwards compatibility to previous version saved config files
+    SnoopTemperatureDeviceTP[0].fill("TEMPERATURE_DEVICE", "Device", temperatureDevice);
+    SnoopTemperatureDeviceTP[1].fill("TEMPERATURE_Property", "Property", temperatureProperty);
+    SnoopTemperatureDeviceTP[2].fill("TEMPERATURE_ATTRIBUTE", "Attribute", temperatureAttribute);
+    SnoopTemperatureDeviceTP.fill(getDeviceName(), "TEMPERATURE_SNOOP", "Temperature Device", OPTIONS_TAB, IP_RW, 0, IPS_IDLE);
 
-    /* Enable/Disable Snoop pickup for Latitude/Longitude and Temperature */
-    EnableSnoopLocationSP[0].fill("ENABLE", "Enable", ISS_OFF);
-    EnableSnoopLocationSP[1].fill("DISABLE", "Disable", ISS_ON);
-    EnableSnoopLocationSP.fill(getDeviceName(), "ENABLE_SNOOP_LOCATION", "Snoop Location", OPTIONS_TAB, IP_RW, ISR_1OFMANY, 0, IPS_IDLE);
+    
+    if (IUGetConfigOnSwitchIndex(getDeviceName(), "ENABLE_SNOOP_TEMPERATURE", &snoopTemperatureIndex) == 0) 
+        setSnoopTemperature = (snoopTemperatureIndex == INDI_ENABLED);
+    
+    // Weather snoop setting for a Weather device ie: Weather station
+    EnableSnoopWeatherSP[0].fill("ENABLE", "Enable", ISS_OFF);
+    EnableSnoopWeatherSP[1].fill("DISABLE", "Disable", ISS_ON);
+    EnableSnoopWeatherSP.fill(getDeviceName(), "ENABLE_SNOOP_WEATHER", "Snoop Weather", OPTIONS_TAB, IP_RW, ISR_1OFMANY, 0, IPS_IDLE);
+    
+    IUGetConfigText(getDeviceName(), "WEATHER_SNOOP", "WEATHER_DEVICE", weatherDevice, MAXINDIDEVICE);
+    IUGetConfigText(getDeviceName(), "WEATHER_SNOOP", "WEATHER_PROPERTY", weatherProperty, MAXINDINAME);
+    IUGetConfigText(getDeviceName(), "WEATHER_SNOOP", "TEMPERATURE_ATTRIBUTE", weatherTempAttribute, MAXINDINAME);
+    IUGetConfigText(getDeviceName(), "WEATHER_SNOOP", "HUMIDITY_ATTRIBUTE", weatherHumidityAttribute, MAXINDINAME);
+    
+    SnoopWeatherDeviceTP[0].fill("WEATHER_DEVICE", "Device", weatherDevice);
+    SnoopWeatherDeviceTP[1].fill("WEATHER_PROPERTY", "Property", weatherProperty);
+    SnoopWeatherDeviceTP[2].fill("TEMPERATURE_ATTRIBUTE", "Temp Attribute", weatherTempAttribute);
+    SnoopWeatherDeviceTP[3].fill("HUMIDITY_ATTRIBUTE", "Humidity Attribute", weatherHumidityAttribute);
+    SnoopWeatherDeviceTP.fill(getDeviceName(), "WEATHER_SNOOP", "Weather Device", OPTIONS_TAB, IP_RW, 0, IPS_IDLE);
```

Comment:
You can simplify this, no need to use IUGetConfigText. Instead, just add this after .fill line

`SnoopWeatherDeviceTP.load();`

This should load the property from config file.

Reply:
Should I be looking to stop using "IUGetConfig..." in general? For instance replacing
```
if (IUGetConfigOnSwitchIndex(getDeviceName(), "ENABLE_SNOOP_TEMPERATURE", &snoopTemperatureIndex) == 0) 
        setSnoopTemperature = (snoopTemperatureIndex == INDI_ENABLED);
```
with something like
```
EnableSnoopTemperatureSP.load();
snoopTemperatureIndex = EnableSnoopTemperatureSP.findOnSwitchIndex();
setSnoopTemperature = snoopTemperatureIndex == INDI_ENABLED;
```